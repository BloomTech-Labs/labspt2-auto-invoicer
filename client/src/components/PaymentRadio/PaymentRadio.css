.payment_checkboxes {
  display: flex;
  align-items: center;
  font-size: 1.4rem;
  margin: 5px 0;
  width: 200px;
}

.payment_checkboxes:hover {
  cursor: pointer;
  color: gray;
}
