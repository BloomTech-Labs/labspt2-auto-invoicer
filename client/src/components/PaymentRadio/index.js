import React from "react";
import PropTypes from "prop-types";

// import styles here
import "./PaymentRadio.css";

const PaymentRadio = props => {
  const { label, handler } = props;
  return (
    <label htmlFor={label} className="payment_checkboxes">
      <input
        type="radio"
        name="payment"
        id={label}
        onChange={handler}
        value={label}
      />
      {props.children}
    </label>
  );
};

PaymentRadio.propTypes = {
  handler: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};

export default PaymentRadio;
