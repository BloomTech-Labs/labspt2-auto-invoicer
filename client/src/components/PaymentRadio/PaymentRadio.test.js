import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import PaymentRadio from "../PaymentRadio";

describe("<PaymentRadio />", () => {
  it("renders PaymentRadio component", () => {
    const { asFragment } = render(<PaymentRadio />);
    expect(asFragment()).toMatchSnapshot();
  });
});
