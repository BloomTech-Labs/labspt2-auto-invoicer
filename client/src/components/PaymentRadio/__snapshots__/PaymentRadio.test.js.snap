// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentRadio /> renders PaymentRadio component 1`] = `
<DocumentFragment>
  <label
    class="payment_checkboxes"
  >
    <input
      name="payment"
      type="radio"
      value=""
    />
  </label>
</DocumentFragment>
`;
